File: Regex/Applications/Valid PAN format.py
import re

regex_pattern = r"^[A-Z]{5}\d{4}[A-Z]$"

n = int(input())
for _ in range(n):
    row_input = input()
    result = "YES" if re.search(regex_pattern, row_input) else "NO"
    print(result)

File: Regex/Applications/Valid PAN format.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <regex>

using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    regex regex_pattern {"^[A-Z]{5}\\d{4}[A-Z]$"};
    int n;
    cin >> n;
    for (int i = 0; i < n; i++) {
        string s;
        cin >> s;
        if (regex_search(s, regex_pattern))
            cout << "YES" << endl;
        else
            cout << "NO" << endl;
    }

    return 0;
}
